### PR TITLE
feat: implement SGT opcode

### LINKED ISSUE #18: Opcode: SGT

### PR DESCRIPTION
**Pull request by Danikim01**
closes #18 
This pr implements the code generation for the SGT opcode.
The tests are not working because i am not able to test with negative numbers

### CODE REVIEW

**Comment by Danikim01** [comment]
tests with signed numbers added

**Review by edg-l — COMMENTED** [review]
please update the readme with the opcode

**Comment by Danikim01** [comment]
> please update the readme with the opcode

@edg-l  readme updated

File: README.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ An EVM-bytecode to machine-bytecode compiler using MLIR and LLVM.
 1. (0x08) ADDMOD
 1. (0x1A) BYTE
 1. (0x50) POP
+1. (0x13) SGT
```
Comment (Oppen) [inline README.md:24]: Nit: keep the order.

File: tests/operations.rs [hunk 2]
```diff
@@ -383,3 +383,77 @@ fn addmod_with_zero_denominator() {
     ];
     run_program_assert_result(program, 0);
 }
+
+#[test]
+fn test_sgt_unsigned_greater_than() {
```
Comment (Oppen) [inline tests/operations.rs:388]: Nit: the numbers are still interpreted as signed.
```suggestion
fn test_sgt_positive_greater_than() {
```

Comment (Oppen) [reply tests/operations.rs:388]: Similar for later uses.

**Review by azteca1998 — APPROVED** [review]

**Pull request merged**